IPython/core/interactiveshell.py

```diff
@@ -3309,12 +3314,54 @@ def error_before_exec(value):
             assert self.history_manager is not None
             # Write output to the database. Does nothing unless
             # history output logging is enabled.
-            self.history_manager.store_output(self.execution_count)
+            hm = self.history_manager
+            hm.store_output(self.execution_count)
+            exec_count = self.execution_count
+
+            if result.result:
+                # Format the result into a MIME bundle
+                try:
+                    mime_obj = (
+                        result.result[0]
+                        if isinstance(result.result, list)
+                        else result.result
+                    )
+                    mime_fig = mime_obj.get_figure()
+                    mime_bundle, _ = self.display_formatter.format(mime_fig)
+                except:
+                    # In case formatting fails, fallback to text/plain
+                    mime_bundle = {"text/plain": repr(result.result)}
```

Comment:
I think we should not expect the formatting to fail; what might fail is calling `.get_figure` if output is not a matplotlib figure, or has this method but is not a matplotlib figure, e.g.:

```python
class MyAccount:
    def get_figure(self, round_to: int):
        """Returns the figure in my account"""
        pass
```